src/components/VHeader/VFilterButton.vue

```diff
@@ -0,0 +1,166 @@
+<template>
+  <VButton
+    v-show="showButton"
+    :variant="button.variant"
+    class="self-center"
+    :pressed="pressed"
+    aria-controls="filter-sidebar"
+    @click="toggleFilters"
+  >
+    <VIcon
+      v-show="icon.show"
+      :icon-path="filterIcon"
+      :class="{ 'me-2': icon.needsPadding }"
+    />
+    <span class="filter-label" :class="button.a11yProps">{{
+      button.label
+    }}</span>
+  </VButton>
+</template>
+
+<script>
+import {
+  computed,
+  defineComponent,
+  onMounted,
+  reactive,
+  useContext,
+  watchEffect,
+} from '@nuxtjs/composition-api'
+import { isScreen } from '~/composables/use-media-query'
+import { useFilterSidebarVisibility } from '~/composables/use-filter-sidebar-visibility'
+
+import filterIcon from '~/assets/icons/filter.svg'
+import VButton from '~/components/VButton.vue'
+import VIcon from '~/components/VIcon/VIcon'
+
+const VFilterButton = defineComponent({
+  name: 'VFilterButton',
+  components: {
+    VIcon,
+    VButton,
+  },
+  props: {
+    /**
+     * Whether the page has been scrolled. If true:
+     * - button has no border.
+     * - on mobile, the button label is blank.
+
+     * @default false
+     */
+    isHeaderScrolled: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  setup(props) {
+    const { i18n, store } = useContext()
+    const isMdScreen = isScreen('md')
+
+    const {
+      isFilterSidebarVisible: pressed,
+      setFilterSidebarVisibility,
+    } = useFilterSidebarVisibility({ mediaQuery: isMdScreen })
+
+    const appliedFilterTags = computed(
+      () => store.getters['search/appliedFilterTags'].length
+    )
+    const isAnyFilterApplied = computed(() => appliedFilterTags.value > 0)
+
+    const labelWithCount = (count) => {
+      return i18n.tc('header.filter-button.with-count', count)
+    }
+    // TODO(obulat): Remove buttonVariants. The aim of this object was to lower
+    // the cognitive load of remembering the variant names when debugging the
+    // button.
+    const buttonVariants = {
+      pressed: 'action-menu-pressed',
+      gray: 'action-menu-muted',
+      bordered: 'action-menu',
+      textOnly: 'action-menu-secondary',
+    }
```

Comment:
Looks like a dangling todo?

Reply:
I was planning to keep it for the work on the assembling the header, and removing after it all works together.